charts/scalardb-cluster/templates/scalardb-cluster/deployment.yaml

```
@@ -64,13 +64,27 @@ spec:
               command:
                 - /usr/local/bin/grpc_health_probe
                 - -addr=:60053
+                {{- if .Values.scalardbCluster.tls.enabled }}
+                - -tls
+                {{- if .Values.scalardbCluster.tls.caRootCertSecret }}
+                - -tls-ca-cert=/tls/certs/ca-root-cert.pem
+                {{- end }}
+                - -tls-server-name={{ .Values.scalardbCluster.tls.overrideAuthority }}
+                {{- end }}
             failureThreshold: 60
             periodSeconds: 5
           livenessProbe:
             exec:
               command:
-              - /usr/local/bin/grpc_health_probe
-              - -addr=:60053
+                - /usr/local/bin/grpc_health_probe
+                - -addr=:60053
+                {{- if .Values.scalardbCluster.tls.enabled }}
+                - -tls
+                {{- if .Values.scalardbCluster.tls.caRootCertSecret }}
+                - -tls-ca-cert=/tls/certs/ca-root-cert.pem
+                {{- end }}
+                - -tls-server-name={{ .Values.scalardbCluster.tls.overrideAuthority }}
```

Don't we need something like this?
```suggestion
                {{- if .Values.scalardbCluster.tls.overrideAuthority }}
                - -tls-server-name={{ .Values.scalardbCluster.tls.overrideAuthority }}
                {{- end }}
```

---

@brfrn169 @feeblefakie  
Ah, I see. You are right!

If the certificate has the hostname `localhost`, we don't need to use the `-tls-server-name` option. I will update it.

Also, I feel the current option `-addr=:60053` (omitting hostname) is a bit confusing. So, I will update it to `-addr=localhost:60053` (specify the `localhost` explicitly) to make it clear.

Thank you for pointing it out!

---

FYI: I was able to run the `grpc_health_probe` command without the `-tls-server-name` option if I use the certificate which has the `localhost` configuration as follows.

* Certificate

```console
$ openssl x509 -in scalardb-cluster.pem -text -noout | grep DNS
                DNS:cluster.scalardb.example.com, DNS:localhost
```

* Run the `grpc_health_probe` command without the `-tls-server-name` option
  * `-addr=:60053` (use `localhost` implicitly)

    ```console
    $ grpc_health_probe -addr=:60053 -tls -tls-ca-cert=/tls/certs/ca-root-cert.pem
    status: SERVING
    ```

  * `-addr=localhost:60053`

    ```console
    $ grpc_health_probe -addr=localhost:60053 -tls -tls-ca-cert=/tls/certs/ca-root-cert.pem
    status: SERVING
    ```

  * `-addr=127.0.0.1:60053` (`127.0.0.1` is not included in the certificate. So, the command failed is expected.)

    ```console
    $ grpc_health_probe -addr=127.0.0.1:60053 -tls -tls-ca-cert=/tls/certs/ca-root-cert.pem
    timeout: failed to connect service "127.0.0.1:60053" within 1s
    ```

  * `-addr=127.0.0.1:60053` and `-tls-server-name localhost` (In case of we need the `-tls-server-name` option.)

    ```console
    $ grpc_health_probe -addr=127.0.0.1:60053 -tls -tls-ca-cert=/tls/certs/ca-root-cert.pem -tls-server-name localhost
    status: SERVING
    ```

---

I fixed it in fbfd0efe8c7a8ca4397e38229a900998e275557a.